packages/client/src/mapping/DataValueMapping.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { Attachment } from "../object/Attachment.js";
+
+/**
+ * Map from the DataValue type to the typescript type that we return
+ */
+export interface DataValueWireToClient {
```

Comment:
By specifying the direction that we intend to be mapping types, it lets us be specific about what we're returning, but more permissive with what we're accepting and sending to the backend. Some of the ClientToWire mappings we can rely on the backend to handle the transformations for us. For example, we can send a number for a long field, and it will be converted to the backend specific type as its parses the request.